src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -48,21 +65,62 @@
 @Description("Specify a window over which functions should be applied. \n" +
   "Supports functions: Rank, Dense Rank, Percent Rank, N tile, Row Number, Median," +
   "Continuous Percentile, Lead, Lag, First, Last, Cumulative distribution, Accumulate.")
-public class WindowAggregation extends SparkCompute<StructuredRecord, StructuredRecord> {
+public class WindowAggregation extends SparkCompute<StructuredRecord, StructuredRecord>
+  implements LinearRelationalTransform {
 
   public static final String NAME = "WindowAggregation";
+  // BigQuery specific aggregations
+  private static final Map<WindowAggregationConfig.Function, String> functionBQSqlMap =
+    new HashMap<WindowAggregationConfig.Function, String>() {{
+      put(WindowAggregationConfig.Function.RANK, "RANK()");
+      put(WindowAggregationConfig.Function.DENSE_RANK, "DENSE_RANK()");
+      put(WindowAggregationConfig.Function.PERCENT_RANK, "PERCENT_RANK()");
+      put(WindowAggregationConfig.Function.N_TILE, "NTILE(%s)");
+      put(WindowAggregationConfig.Function.ROW_NUMBER, "ROW_NUMBER()");
+      put(WindowAggregationConfig.Function.MEDIAN, "PERCENTILE_CONT(%s, 0.5)");
+      put(WindowAggregationConfig.Function.CONTINUOUS_PERCENTILE,
+        "PERCENTILE_CONT(%s,%s) OVER()");
+      put(WindowAggregationConfig.Function.DISCRETE_PERCENTILE,
+        "PERCENTILE_DISC(%s,%s) OVER()");
+      put(WindowAggregationConfig.Function.LEAD, "LEAD(%s)");
+      put(WindowAggregationConfig.Function.LAG, "LAG(%s)");
+      put(WindowAggregationConfig.Function.FIRST, "FIRST_VALUE(%s)");
+      put(WindowAggregationConfig.Function.LAST, "LAST_VALUE(%s)");
+      put(WindowAggregationConfig.Function.CUMULATIVE_DISTRIBUTION, "CUME_DIST()");
+      put(WindowAggregationConfig.Function.ACCUMULATE, "SUM(%s)");
+    }};
+  private static final Logger LOG = LoggerFactory.getLogger(WindowAggregation.class);
   private final WindowAggregationConfig config;
   private Schema outputSchema;
 
-  private static final Logger LOG = LoggerFactory.getLogger(WindowAggregation.class);
-
   public WindowAggregation(WindowAggregationConfig config) {
     this.config = config;
   }
 
+  private static Optional<ExpressionFactory<String>> getExpressionFactory(RelationalTranformContext ctx,
+                                                                          WindowAggregationConfig config,
+                                                                          FailureCollector failureCollector) {
+    List<WindowAggregationConfig.FunctionInfo> functionInfos = config.getAggregates(failureCollector);
+    boolean requiresBigQueryCapability = false;
```

Comment:
This value will always be true, unless the window aggregation has no operations? does this make sense?

Reply:
Yes.Reference to Group By Aggregator.

Reply:
You are not using the `requiresBigQueryCapability` for anything, please remove.